### PR TITLE
CRUD Operations

### PR DESCRIPTION
**Pull request by K-Ds**

### CODE REVIEW

File: Server/src/controllers/postControllers.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import postServices from "../services/postServices";
+import postValidator from "../validators/postValidator";
+
+const postController = {};
+
+// Get all posts
+postController.getAllPosts = async (req, res) => {
```
Comment (habaumugisha1) [inline Server/src/controllers/postControllers.js:7]: don't use this object here

File: Server/src/controllers/postControllers.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  const posts = await postServices.getAll();
+  return res.json(posts);
+};
+
+// Get a post by its ID
+postController.getPost = async (req, res) => {
+  const postId = req.params.id;
+  try {
+    const post = await postServices.getOnePost(postId);
+    return res.json(post);
```
Comment (habaumugisha1) [inline Server/src/controllers/postControllers.js:17]: set status on a response

File: Server/src/models/Comment.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const mongoose = require("mongoose");
+
+const commentSchema = new mongoose.Schema({
+  content: { type: String },
+  user: { type: String },
```
Comment (habaumugisha1) [inline Server/src/models/Comment.js:5]: where is a post to be commented on!

Comment (K-Ds) [reply Server/src/models/Comment.js:5]: the yet done with the comments and like functions

**Review by habaumugisha1 — COMMENTED** [review]
Good improvement and good work, find the attached feedback

File: Server/src/models/Comment.js (hunk 3 above)
Comment (habaumugisha1) [reply Server/src/models/Comment.js:5]: this is a model for comment

File: Server/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "server",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "start": "babel-node src/index.js"
```
Comment (habaumugisha1) [inline Server/package.json:8]: You can also check how they configured script for production

File: Server/.env [hunk 5]
```diff
@@ -0,0 +1,3 @@
+CLOUD_NAME = "k-ds"
+CLOUDINARY_API_KEY = "823114164825559"
```
Comment (Diane10) [inline Server/.env:2]: don't push the value

[the remark this answers is not in the document]
Comment (K-Ds) [reply Server/.env:2]: That was just provisional because am still learning how to use the config files, but by today it will be changed

**Comment by Diane10** [comment]
Woww let me know if you have done I ask the money

File: Server/src/controllers/postControllers.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+// Create new post
+export const newPost = async (req, res) => {
+  let imgUrl = "";
+  try {
+    const cloudinaryResult = await cloudinary(req.body.img);
+    imgUrl = cloudinaryResult.url;
+  } catch (err) {
+    return res.status(500).json({ error: err });
```
Comment (habaumugisha1) [inline Server/src/controllers/postControllers.js:33]: return err.message instead of err

File: Server/src/controllers/postControllers.js [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+};
+
+// Update a post
+export const updatePost = async (req, res) => {
+  let imgUrl = "";
+  try {
+    const cloudinaryResult = await cloudinary(req.body.img);
+    imgUrl = cloudinaryResult.url;
+  } catch (err) {
+    return res.status(500).json({ error: err });
```
Comment (habaumugisha1) [inline Server/src/controllers/postControllers.js:57]: the same here!

File: Server/src/middleware/auth.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+function auth(req, res, next) {
+  const token = req.header("X-Auth-Token");
+
+  if (!token) {
+    return res.status(401).json({ error: "Access denied, No token provided." });
+  }
+
+  try {
+    const decoded = jwt.verify(token, "jwtPrivateKey");
```
Comment (habaumugisha1) [inline Server/src/middleware/auth.js:11]: hide a secret in a variable environment

File: Server/src/models/User.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    unique: true,
+  },
+  password: { type: String, required: true, minlength: 5, maxlength: 1024 },
+  isAdmin: { type: Boolean, required: true, default: false },
+});
+
+userSchema.methods.generateToken = function () {
+  const token = jwt.sign(
+    { _id: this._id, isAdmin: this.isAdmin },
+    "jwtPrivateKey"
```
Comment (habaumugisha1) [inline Server/src/models/User.js:22]: hide the secret, "jwtPrivateKey"

**Pull request merged**